### PR TITLE
fix doc permissions example and other docs linting

### PR DESCRIPTION
**Pull request by fmigneault**
## Changes

Fix the permissions example in [Ouranosinc/Magpie@fix-docs-example/docs/permissions.rst](https://github.com/Ouranosinc/Magpie/compare/fix-docs-example?expand=1#diff-9cdf52695210be488a07bb9d41acd2adec865440097187b6ffdf3a1cdff78465)

### CODE REVIEW

File: docs/permissions.rst [hunk 1]
```diff
@@ -744,60 +744,71 @@ are as follows:
...
+from its parents ``resource-B1`` (through inherited :term:`Group` :term:`Permission <Permissions>`)
+and ``service-3`` (through direct :term`User` :term:`Permission <Permissions>`).
+This demonstrates why, although |effective_permissions|_ imply |inherited_permissions|_,
+they do not necessarily resolve to the same result according to the effective :term:`Resource` hierarchy
+and its parent-children resolution implementation.
 
 Using ``effective`` query tells `Magpie` to rewind the :term:`Resource` tree from the requested :term:`Resource` up to
 the top-most :term:`Service` in order to accumulate all |inherited_permissions|_ observed along the way for every
-encountered element. All :term:`Permission` that is applied *higher* to the requested :term:`Resource` are considered
+encountered element. All :term:`Permission` that is applied *higher* that the requested :term:`Resource` are considered
```
Comment (ChaamC) [inline docs/permissions.rst:811]: I would maybe correct this :
All :term:\`Permission\<Permissions\>\` that are applied \*higher\* **than** the ...

File: docs/permissions.rst [hunk 2]
```diff
@@ -744,60 +744,71 @@ are as follows:
...
+    /users/example-user/resources/service-2/permissions?inherited=true      => [write]  (2)
     /users/example-user/resources/resource-A/permissions?inherited=true     => [read]
     /users/example-user/resources/service-3/permissions?inherited=true      => [write]
-    /users/example-user/resources/resource-B1/permissions?inherited=true    => [read]   (1)
+    /users/example-user/resources/resource-B1/permissions?inherited=true    => [read]   (2)
     /users/example-user/resources/resource-B2/permissions?inherited=true    => []
 
 As illustrated, requesting for |inherited_permissions|_ now also returns :term:`Group`-related :term:`Permission`
-:sup:`(1)` where they where not returned before with only :term:`User`-related :term:`Permission`.
+:sup:`(1)` that where not returned before with only :term:`User`-related :term:`Permission`. Note that returned
```
Comment (ChaamC) [inline docs/permissions.rst:777]: I think the (1) here should be (2)?

File: docs/permissions.rst [hunk 3]
```diff
@@ -744,60 +744,71 @@ are as follows:
...
+For simplification purposes, we will use the names directly in following steps, but remember that actual :term:`API`
+requests would normally require unique identifiers for :term:`Resource` resolution. Lets observe what happens using
+different query parameters with request ``GET /users/{user_name}/resources/{resource_id}/permissions``.
 
 If no query parameter is specified, we obtain permissions as follows:
 
 .. code-block::
 
-    /users/example-user/resources/service-1/permissions                     => [write]
+    /users/example-user/resources/service-1/permissions                     => [write]  (1)
```
Comment (ChaamC) [inline docs/permissions.rst:755]: Should the (1) here be referenced in the text right below?

File: docs/permissions.rst [hunk 4]
```diff
@@ -744,60 +744,71 @@ are as follows:
...
 
 .. code-block::
 
-    /users/example-user/resources/service-1/permissions?effective=true      => [write]
+    /users/example-user/resources/service-1/permissions?effective=true      => [write]          (1)
     /users/example-user/resources/service-2/permissions?effective=true      => [write]          (2)
     /users/example-user/resources/resource-A/permissions?effective=true     => [read, write]    (3)
-    /users/example-user/resources/service-3/permissions?effective=true      => []
+    /users/example-user/resources/service-3/permissions?effective=true      => [write]          (1)
     /users/example-user/resources/resource-B1/permissions?effective=true    => [read]           (2)
```
Comment (ChaamC) [inline docs/permissions.rst:788]: Doesn't `resource-B1` also inherit the `write` permission from the `service-3`, like the `resource-B2`?

**Review by ChaamC — CHANGES_REQUESTED** [review]
Thank you for reviewing the docs!
Some minor comments and it should be good to go.

File: docs/permissions.rst (hunk 4 above)
Comment (fmigneault) [reply docs/permissions.rst:788]: You're right!

**Comment by fmigneault** [comment]
@ChaamC Thanks for the review!

**Pull request merged**